doc/design_decisions.rst

```diff
@@ -0,0 +1,87 @@
+Design decisions
+================
+
+
+Single Pool instance
+--------------------
+
+We have decided to keep all packages incl. comps in a single Pool instance.
+Lookups are slower due to iterating through a larger data set
+(using smaller dedicate Pool instances per package type would be faster),
+but we don't have to duplicate repo definitions, keep the repos in sync
+across all Pool instances etc.
+
+Modularity implementation is an exception to this rule
+because the modulemd records require special treatment.
```

Comment:
I think the reason is actually stronger than what the above suggests, I'd phrase it like:
> Note there is a second libsolv pool in implementation of modularity, which is used to resolve module dependencies. This is a separate resolution namespace from the main libsolv pool, unrelated to and incompatible with rpm packages, and needs to be kept separate.

Reply:
Thanks, this makes more sense than my original text.